plugins/org.jkiss.dbeaver.ext.mysql.ui/src/org/jkiss/dbeaver/ext/mysql/ui/config/MySQLCommandChangeUser.java

```diff
@@ -86,27 +101,36 @@ public void afterExecute(DBCSession session, Throwable error)
                         }
                     }
                 });
+        } else {
+            addAction(actions, this::generateRenameScript);
         }
-        boolean hasSet;
         final MySQLDataSource dataSource = getObject().getDataSource();
         if (MySQLUtils.isAlterUSerSupported(dataSource)) {
-            StringBuilder script = new StringBuilder();
-            if (generateAlterScript(script)) {
-                actions.add(new SQLDatabasePersistAction(MySQLUIMessages.edit_command_change_user_action_update_user_record, script.toString()));
-            }
+            addAction(actions, this::generateAlterScript);
         } else {
-            String updateSQL = generateUpdateScript();
-            if (updateSQL != null) {
-                actions.add(new SQLDatabasePersistAction(MySQLUIMessages.edit_command_change_user_action_update_user_record, updateSQL));
-            }
-            updateSQL = generatePasswordSet();
-            if (updateSQL != null) {
-                actions.add(new SQLDatabasePersistAction(MySQLUIMessages.edit_command_change_user_action_update_user_record, updateSQL));
-            }
+            addAction(actions, this::generateUpdateScript);
+            addAction(actions, this::generatePasswordSet);
         }
         return actions.toArray(new DBEPersistAction[0]);
     }
 
+    private static void addAction(@NotNull Collection<? super DBEPersistAction> actions, @NotNull Supplier<String> sqlGenerator) {
+        String sql = sqlGenerator.get();
+        if (sql != null) {
+            actions.add(new SQLDatabasePersistAction(MySQLUIMessages.edit_command_change_user_action_update_user_record, sql));
+        }
+    }
+
+    @Nullable
+    private String generateRenameScript() {
+        MySQLUser mySQLUser = getObject();
+        if (mySQLUser.isPersisted() && (!Objects.equals(mySQLUser.getUserName(), mySQLUser.getPersistedUserName()) || !Objects.equals(mySQLUser.getHost(), mySQLUser.getPersistedHost()))) {
+            return "RENAME USER " + mySQLUser.getPersistedFullName() + " TO " + getObject().getFullName() + ";"; //$NON-NLS-1$ //$NON-NLS-3$
```

Comment:
Missing NLS marker for the second string

Reply:
Woops.